ext/gd/gd.c

```diff
@@ -1861,15 +1857,66 @@ PHP_FUNCTION(imagexbm)
 /* {{{ Output GIF image to browser or file */
 PHP_FUNCTION(imagegif)
 {
-	_php_image_output_ctx(INTERNAL_FUNCTION_PARAM_PASSTHRU, PHP_GDIMG_TYPE_GIF, "GIF");
+	zval *imgind;
+	gdImagePtr im;
+	gdIOCtx *ctx;
+	zval *to_zval = NULL;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "O|z!", &imgind, gd_image_ce, &to_zval) == FAILURE) {
+		RETURN_THROWS();
+	}
```

Comment:
Is the `to_zval` a by-ref param?

Reply:
No it's not, but it can be either null, a stream, or a string being the file name to write the image to.